### ISSUE TITLE
Indentation with JavaScript

### ISSUE THREAD

**Issue by alizahid** [adobe/brackets#10512]
Hi. I have a slightly different style of writing code than the "standard" way, as seen in this [video](https://www.dropbox.com/s/b43mtvcj85m21av/my-style.mov?dl=0).

	$.ajax('url.php',
	{
		var: value,
		var: value
		
	}).then(function()
	{
		// what?
		
	}, function()
	{
		// hello

	});

If you watch the video, you'll notice the indentation problems that I run into. Is there a setting I can tweak to fix this? It really kills my flow and messes with my OCD big-time.

For reference, here's a [video](https://www.dropbox.com/s/pr39cb1pd4qguov/normal.mov?dl=0) of the more popular / standard of coding.

Thanks.

**Comment by redmunds** [comment]
@alizahid Take a look at the [list of supported preferences](https://github.com/adobe/brackets/wiki/How-to-Use-Brackets#list-of-supported-preferences). I think setting `"smartIndent": false` will fix that.

**Comment by alizahid** [comment]
It has, thank you!

Also, when I insert a few lines, they're all indented to my cursor's current position but after I insert another line, they don't clear up. [This video](https://dl.dropboxusercontent.com/u/7401989/videos/indent.mov) illustrates what I mean.

Is there a way to deal with that? E.g., Android Studio and Sublime Text add those tabs but remove them if I press `enter` again to move to the next line.

**Comment by redmunds** [comment]
@alizahid No, there's currently no way to do that in Brackets. That would be nice. I think Visual Studio also does that. Update the Title and Description of this Issue to describe the desired behavior for this Feature Request.

**Comment by alizahid** [comment]
It would definitely be nice. I'm used to it, and it's kinda annoying not having it.

I've updated the title + description. Did I do that correctly?

**Comment by redmunds** [comment]
@alizahid I think this looks good.

This needs to be a preference, because I know some people would not want this.

You only want this to clear indentation that you recently inserted (not *any* indentation), right? If so, that might be tricky to get right.

**Comment by ChamuMutezva** [comment]
Visual studio has made life easier. You can simply right click on an open file and select "Format Document", its not properly indented.
